src/cryptonote_core/service_node_list.cpp

```diff
@@ -505,6 +508,7 @@ namespace service_nodes
       {
         m_rollback_events.pop_front();
       }
+      m_rollback_events.push_front(std::unique_ptr<rollback_event>(new prevent_rollback(cull_height)));
```

Comment:
No need to do this. ROLLBACK_EVENT_EXPIRATION_BLOCKS must be greater than reorg safety buffer (at least, it should be). We assume that there will never be a reorg greater than rollback event expiration blocks.

Reply:
(please do take this part out though, it's not logical)

Reply:
We don't want to guard our code? Reorgs greater than 20 have been happening in testnet. Mainnet sure, might not happen. If it triggers, then everyones chain halts.